File: student/src/main/java/com/ktxdevelopment/model/entity/Grade.java
package com.ktxdevelopment.model.entity;

import jakarta.persistence.*;
import lombok.*;


@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@RequiredArgsConstructor
@Entity(name = "grade")
@Table(name = "grades")
class Grade {

    @Id
    @GeneratedValue
    private Integer id;

    @OneToOne
    private Student student;

    @OneToOne
    private Transcript transcript;

    @OneToMany(mappedBy = "courseId")
    private Course course;

    private Double score;
}

File: student/src/main/java/com/ktxdevelopment/model/entity/Grade.kt
package com.ktxdevelopment.model.entity

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.Table
import lombok.AllArgsConstructor
import lombok.NoArgsConstructor


//@Entity
@Table
data class Grade(
    @Id
    @GeneratedValue
    val id: Int,
    val studentId: Int,
    val courseId: Int,
    val grade: Double
)
